src/core/utils.js

```diff
@@ -1,8 +1,9 @@
 'use strict';
 
-function getMantissaDecimalString(bignum) {
-  let mantissa = bignum.toPrecision(16)
-    .replace(/\./, '')      // remove decimal point
+// Should only be used directly when the string is from a BigNumber
```

Comment:
I don't think it matters if the string is from a BigNumber. Parameter name could be `decimalString` maybe? Any decimal-encoded numeric string should be the same right? Also, I don't think the function requires 16 digits of precision. It might be good to add a case for if `mantissa.length` is greater than 16 though so that it works for all inputs.